tests/base/test_regularization.py

```diff
@@ -334,8 +335,8 @@ def test_update_of_sparse_norms(self):
         v = np.random.rand(mesh.nC)
 
         cell_weights = np.random.rand(mesh.nC)
-
-        reg = regularization.Sparse(mesh, weights=cell_weights)
+        reg = regularization.Sparse(mesh)
+        reg.set_weights(cell_weights=cell_weights)
```

Comment:
Small change here. I prefer keep passing the weights through the constructor in this case.
```suggestion
        reg = regularization.Sparse(mesh, weights={"cell_weights": cell_weights})
```